specs/capri/object_lifecycle_management.md

```diff
@@ -0,0 +1,568 @@
+
+### When to use Lifecycle Configuration?
+Define lifecycle configuration rules for objects that have a well-defined lifecycle. For example:
+- If you upload periodic logs to a bucket, your application might need them for a week or a month. After that, you might want to delete them.
+- Some documents are frequently accessed for a limited period of time. After that, they are infrequently accessed. At some point, you might not need real-time access to them, but your organization or regulations might require you to archive them for a specific period. After that, you can delete them.
+- You might upload some types of data to the cloud primarily for archival purposes. For example, you might archive digital media, financial and healthcare records, raw genomics sequence data, long-term database backups, and data that must be retained for regulatory compliance.
+
+### Objectives
+* Goal : Develop OpenSDS object lifecycle management mechanism. Allow tenants to manage lifecycle configuration policies through APIs. There are two scenarios in object lifecycle management.
+	* Scenario 1 - Object lifecycle management across different cloud service providers and support of choosing the storage class while setting the lifecycle rules.	
+Example: If there is a bucket in STANDARD storage class of AWS S3 and it needs to be    transitioned to COOL storage class of Azure Blob we can set a rule which will do it automatically.
+
+	*  Scenario 2 - Object lifecycle management within the cloud storage. 
+Example : Transitioning a bucket from S3 “STANDARD_IA”  to S3 “GLACIER” using lifecycle rule defined in OpenSDS. 
+
+For both the scenarios we must have APIs defined which can read the lifecycle rule and perform lifecycle operations on the OpenSDS bucket. 
+* Sub-Goal 1 : Provide lifecycle configuration management for  non-versioned bucket.
+* Sub-Goal 2 : Provide lifecycle configuration management for versioning enabled/suspended bucket (This is not supported currently, but in the future).
+
+## Solution
+### Defining storage tiers in OpenSDS
+
+Compared with storage class defined in AWS s3, OpenSDS use storage tier internally, and there is a mapping between storage class and storage tier. 
+
+OpenSDS s3 API is compatible with AWS s3, that means user can use storage class when they use the s3 API, OpenSDS will convert storage class to specific storage tier in background.
+
+As default, there are three tiers, Tier_1, Tier_99 and Tier_999, and the mapping between storage tier and class is showed as below:
+
+| | Tier_1 | Tier_99 | Tier_999 |
+|---------------|---------------|---------------|---------------|
+| AWS S3 | STANDARD | STANDARD_IA | GLACIER |
+| Azure Blob | HOT | COOL | ARCHIVE |
+| HW OBS | STANDARD | WARM | COLD |
+| GCP | Multi-Regional | NearLine | ColdLine |
+| Ceph S3 | STANDARD | - | - |
+| Fusion Storage Object | STANDARD | - | - |
+
+Notes:
```

Comment:
Add a note that IBM Cloud Storage is not supported in object lifecycle management.